compiler/rustc_middle/src/mir/consts.rs

```diff
@@ -399,101 +409,6 @@ impl<'tcx> Const<'tcx> {
         Self::Val(val, ty)
     }
 
-    /// Literals are converted to `Const::Val`, const generic parameters are eagerly
-    /// converted to a constant, everything else becomes `Unevaluated`.
-    #[instrument(skip(tcx), level = "debug", ret)]
-    pub fn from_anon_const(
```

Comment:
*Thank you* so much for removing this... I never had any clue what it really does, it was such a complicated mess, but it's all sorts of stuff I didn't understand so I was not able to clean it up.